app/Http/Controllers/CommentController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Contracts\Services\CommentServiceInterface;
+use App\Http\Requests\StoreCommentRequest;
+use App\Models\Comment;
+use Illuminate\View\View;
+use Illuminate\Support\Facades\Auth;
+use App\Models\Posts;
+use Illuminate\Http\RedirectResponse;
+
+class CommentController extends Controller
+{
+    protected $commentService;
+
+    public function __construct(CommentServiceInterface $commentService){
+        $this->commentService = $commentService;
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param StoreCommentRequest $request
+     * @return string
+     */
+    public function store(Request $request)
+    {
+        $validatedData = $request->validate([
+            'post_id' => 'required|integer', // Validate the post_id
+            'comment' => 'required|string',
+        ]);
+    
+        $commentData = [
+            'user_id' => auth()->id(),
+            'comment' => $validatedData['comment'],
+            'posts_id' => $validatedData['post_id'], // Use the post_id from the form data
+        ];
+    
+        // Create the comment with the correct posts_id
+        Comment::create($commentData);
```

Comment:
Don't do this in controller. 
Please don't repeat same error.
Be careful next time.